### PR TITLE
refactor(TSCH-37): space between navbar and search box

### PR DESCRIPTION
**Pull request by amaalej**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/Talend/bootstrap-theme/blob/master/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
With the former mockup, there was space between the navbar and the search bar.

**What is the new behavior?**
I reduced that space to correspond the new mockup.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
I omitted the icons, as they will be managed by the search box component

### CODE REVIEW

File: example/index.html [hunk 1]
```diff
@@ -102,10 +102,6 @@ <h1 id="navbar">Navbar</h1>
                       <div class="form-group">
                         <input type="text" class="form-control" aria-label="Search over the whole app" placeholder="Search">
                       </div>
-                      <button type="submit" class="btn btn-link">
```
Comment (frassinier) [inline example/index.html]: Do you consider a `form` without `type="submit"`? I'm not.

Comment (jmfrancois) [inline example/index.html]: but it's a form ?

Comment (amaalej) [inline example/index.html]: done.

File: example/index.html [hunk 2]
```diff
@@ -102,10 +102,7 @@ <h1 id="navbar">Navbar</h1>
...
-                        <i class="fa fa-search" aria-hidden="true" title="Search"></i>
-                        <span class="sr-only">Search</span>
-                      </button>
+                      <button type="submit" class="btn btn-link sr-only"/>
```
Comment (frassinier) [inline example/index.html]: I will be painful but why did you remove button content?

```html
<button type="submit" class="sr-only">Search</button>
```

File: example/index.html [hunk 3]
```diff
@@ -149,7 +146,7 @@ <h1 id="navbar">Navbar</h1>
                       <div class="form-group">
                         <input type="text" class="form-control" placeholder="Search">
                       </div>
-                      <button type="submit" class="btn btn-default">Submit</button>
+                      <button type="submit" class="btn btn-link sr-only"/>
```
Comment (frassinier) [inline example/index.html]: Same here

```html
<button type="submit" class="sr-only">Submit</button>
```

File: src/theme/_navbar.search.scss [hunk 4]
```diff
@@ -18,20 +18,8 @@ $navbar-search-btn-width: 40px !default;
 			padding-right: $navbar-search-btn-width;
 			font-style: italic;
 			color: $navbar-search-btn-color;
-			background: none;
```
Comment (frassinier) [inline src/theme/_navbar.search.scss]: Be careful, text color is white by default here.

File: src/theme/_navbar.search.scss [hunk 5]
```diff
@@ -18,20 +18,8 @@ $navbar-search-btn-width: 40px !default;
 			padding-right: $navbar-search-btn-width;
 			font-style: italic;
 			color: $navbar-search-btn-color;
```
Comment (jsomsanith) [inline src/theme/_navbar.search.scss]: The color is white on white background

File: src/theme/_navbar.search.scss (hunk 4 above)
Comment (amaalej) [inline src/theme/_navbar.search.scss]: Indeed, I will set input color property.

**Pull request merged**